yt/python/yt/wrapper/client_impl.py

```diff
@@ -2496,6 +2496,36 @@ def run_operation(
             client=self,
             sync=sync, run_operation_mutation_id=run_operation_mutation_id, enable_optimizations=enable_optimizations)
 
+    def run_query(
+            self,
+            engine, query,
+            settings=None, files=None, stage=None, annotations=None, access_control_objects=None,
+            sync=True):
+        """
+        Run query and track its progress (unless sync = false).
+
+        :param engine: one of "ql", "yql", "chyt", "spyt".
+        :type engine: str
```

Comment:
we can use typehints now (and trying to use them in new code)

Reply:
Annotated new code with type hints except for `client=None` argument which requires tricky handling (we cannot import the file that defines YtClient, as it would introduce a circular import, so a metaclass should be defined separately which is definitely out of scope for this PR)